UnityProject/Assets/Scripts/Systems/StationObjectives/Objectives/ShipResources.cs

```diff
@@ -0,0 +1,140 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using System.Linq;
+using System.Text;
+using Strings;
+using Systems.Cargo;
+
+namespace StationObjectives
+{
+	/// <summary>
+	/// An objective to ship items to nanotrasen
+	/// </summary>
+	[CreateAssetMenu(menuName = "ScriptableObjects/StationObjectives/ShipResources")]
+	public class ShipResources : StationObjective
+	{
+		/// <summary>
+		/// The pool of possible resources to ship
+		/// </summary>
+		[SerializeField]
+		private ItemDictionary ItemPool = null;
+
+		/// <summary>
+		/// The resource to ship
+		/// </summary>
+		private string ItemName;
+
+		/// <summary>
+		/// The number of items needed to ship to complete the objective
+		/// </summary>
+		private int Amount;
+
+		/// <summary>
+		/// Current amount of the item sold.
+		/// </summary>
+		private int AmountSold = 0;
+
+		protected override bool CheckCompletion()
+		{
+			var finalReport = new StringBuilder(victoryDescription);
+			finalReport.Replace("SHIPPEDVAL", $"{AmountSold}");
+			victoryDescription = finalReport.ToString();
+			Logger.Log($"amount sold {AmountSold}");
+			Logger.Log($"amount {Amount}");
+			if (AmountSold >= Amount)
+			{
+				return true;
+			}
+			else
+			{
+				return false;
+			}
+		}
+		private void OnEnable()
+		{
+			EventManager.AddHandler(EVENT.ItemSold, CheckItemSold);
+		}
+		private void OnDisable()
+		{
+			EventManager.RemoveHandler(EVENT.ItemSold, CheckItemSold);
+		}
+		protected override void Setup()
+		{
+			var possibleItems = ItemPool.ToList();
+
+			if (possibleItems.Count == 0)
+			{
+				Logger.LogWarning("Unable to find any items to ship. This shouldn't happen!");
+			}
+
+			var itemEntry = possibleItems.PickRandom();
+			if (itemEntry.Key == null)
+			{
+				Logger.LogError($"Objective failed because the item type chosen is somehow destroyed." +
+								" Definitely a programming bug. ", Category.Round);
+				return;
+			}
+
+			ItemName = itemEntry.Key.Item().InitialName;
+
+			if (string.IsNullOrEmpty(ItemName))
+			{
+				Logger.LogError($"Objective failed because the InitialName has not been" +
+								$" set on this objects ItemAttributes. " +
+								$"Item: {itemEntry.Key.Item().gameObject.name}", Category.Round);
+				return;
+			}
+
+			Amount = Random.Range(itemEntry.Value - itemEntry.Value / 3, itemEntry.Value + itemEntry.Value / 3);
```

Comment:
a comment to explain the range here would be handy.

Reply:
good point, i'll do that